Fix propel datacollector

ldaspt opened the pull request:
Fix error in profiler on the Propel panel

```
An exception has been thrown during the rendering of a template ("Propel\Bundle\PropelBundle\DataCollector\PropelDataCollector::getQueries(): Return value must be of type array, Symfony\Component\VarDumper\Cloner\Data returned").
```

Commit:
Fix propel datacollector

Fix error in profiler on the Propel panel

```
An exception has been thrown during the rendering of a template ("Propel\Bundle\PropelBundle\DataCollector\PropelDataCollector::getQueries(): Return value must be of type array, Symfony\Component\VarDumper\Cloner\Data returned").
```

ldaspt commented:
@SkyFoxvn Friendly ping

SkyFoxvn commented:
sorry about the delay
i just checked that but im not sure thats the solution sure maybe it will work depending of few thinks for example if we are working with single tone PropelDataCollector but if we are making new instance of this class you will get printed only the instance queries not all queries as in the previous case ussing logger->getQueries

**also another think is if you check the logger class PropelLogger -> getQueries you will easy see that function cant return anything else then array so the problem is somewere else** please provide me with more information so i can reproduce the problem and find what happens 

NOTE: i will look it after the new year but still i will need more info to reproduce it

Commit:
Revert "Fix propel datacollector"

This reverts commit bd46da6fdcc4b65a632dbe01c8f1c077af54dbd4.

Commit:
Fix propel datacollector

Fix error in profiler on the Propel panel

```
An exception has been thrown during the rendering of a template
("Propel\Bundle\PropelBundle\DataCollector\PropelDataCollector::getQueries(): Return value
must be of type array, Symfony\Component\VarDumper\Cloner\Data returned").
```

Related to adding hint type to the medthod in the commit
https://github.com/SkyFoxvn/PropelBundle/commit/58fa12e924d8cd4ce18a712571b4c68de38d5521#diff-53d4d12dd850e3a771f9af985f2fea723cbbaee3f0d3509da76ab67404a0c05b

ldaspt commented:
@SkyFoxvn Sorry, I edited the wrong line, You're right. I updated the PR. 

I believe the regression comes from adding the hint type to the "getQueries" method in the commit https://github.com/SkyFoxvn/PropelBundle/commit/58fa12e924d8cd4ce18a712571b4c68de38d5521#diff-53d4d12dd850e3a771f9af985f2fea723cbbaee3f0d3509da76ab67404a0c05b

You can reproduce it by going to the profiler panel, you should have the error `An exception has been thrown during the rendering of a template ("Propel\Bundle\PropelBundle\DataCollector\PropelDataCollector::getQueries(): Return value must be of type array, Symfony\Component\VarDumper\Cloner\Data returned").`

SkyFoxvn commented:
@ldaspt Hi
i was able to reproduce the problem and see what happens.
Look like your solution it will work but im not sure if it can brake something else and we dont really need to convert the data to array. We just need to change the return type to the right one and all will work as supposed.

**On the top of the script(line 18) just add:**
`use Symfony\Component\VarDumper\Cloner\Data;`
(which is the class created by $this->cloneVar()) 

**then change(line 60):**
`"public function getQueries(): Array" to "public function getQueries(): Data"`

Commit:
Propel datacollector - return hint as Data for getQueries

ldaspt commented:
@SkyFoxvn Ok with your solution, I fixed it

SkyFoxvn commented:
tnx for the help
